authors/views.py

```diff
@@ -56,3 +64,71 @@ def post(self, request, author_id):
             author.update_fields_with_request(request)
             return Response(serializer.data)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class InboxListView(ListCreateAPIView):
+    """
+    POST to inbox: a foreign server sends some json object to the inbox. server basic auth required
+    GET from inbox: get all objects for the current user. user jwt auth required
+    """
+    # permission_classes = [permissions.permissions.IsAuthenticated]
+    serializer_class = InboxObjectSerializer
+
+    def get_serializer_context(self):
+        # append the endpoint author to the serializer context
+        # the serializer will use it for creating an InboxObject.
+        try:
+            author = Author.objects.get(id=self.kwargs.get('author_id'))
+        except:
+            raise exceptions.NotFound
+        return {
+            'author': author
+        }
+
+    def get_queryset(self):
+        try:
+            author = Author.objects.get(id=self.kwargs.get('author_id'))
+        except:
+            raise exceptions.NotFound
+        return author.inbox_objects.all()
+
+
+@api_view(['POST'])
+# @permission_classes([permissions.IsAuthenticated])
+def internally_send_friend_request(request, author_id, foreign_author_url):
+    """
+    the /author/<author_id>/friend_request/<foreign_author_url>/ endpoint
+    - author_id: anything other than slash, but we hope it's a uuid
+    - foreign_author_url: anything, but we hope it's a valid url.
+
+    used only by local users, jwt authentication required.
+    Its job is to fire a POST to the foreign author's inbox with a FriendRequest json object.
+
+    NOTE: I think putting url inside a url sucks, too. -Lucas
+
+    questions:
+    - what to expect from POST result?
+    - How do we know the friend request has been accepted?
+    """
+    import requests
+
+    try:
+        author = Author.objects.get(id=author_id)
+    except:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+    # get that foreign author's json object first
+    foreign_author_json = requests.get(foreign_author_url).json()
+
+    # TODO do we check for foreign author validity?
```

Comment:
maybe we can do a GET request to the <foreign_author_url> and see if it returns 200?